src/entities/project.ts

```diff
@@ -341,13 +351,38 @@ class Project extends BaseEntity {
 
   // Status 7 is deleted status
   mayUpdateStatus(user: User) {
-    if (this.statusId === ProjStatus.cancel) return false;
+    if (this.statusId === ProjStatus.cancel) {
+      throw new Error(
+        errorMessages.THIS_PROJECT_IS_CANCELLED_OR_DEACTIVATED_ALREADY,
+      );
+    }
 
     if (this.users.filter(o => o.id === user.id).length > 0) {
```

Comment:
I have a faint memory this works on fairly recent projects, but old really projects won't. It seems they don't have the user relation set up correctly or non existant, but they have admin string. This was something that happened months ago, mostly giveth projects.
I think we should use the filter as is but also add an OR  for checking the admin string which is also important.
```
if (this.users.filter(o => o.id === user.id).length > 0 || user.id === Number(this.admin))
```

Reply:
Or just the admin string... which honestly has the source of truth, without filtering a list of users.

Reply:
Nice catch, I have changed the condition